docs/INTEGRATION_FOREGROUND_FILTERS.md

```diff
@@ -0,0 +1,123 @@
+
+Foreground filters are only usable with a SearchController.
+
+In the simplified example below, a foreground filter is used to pre-apply a filter for the `on_sale` field. 
+
+```typescript
+const config = {
+	client: {
+		globals: {
+			siteId: 'abc123',
+		},
+	},
+	controllers: {
+		search: [
+			{
+				url: {
+					initial: {
+						parameters: {
+							filter: {
+								state: {
+									on_sale: ['yes'],
+								}
+							},
+						},
+					}
+				},
+				config: {
+					id: 'search',
+				},
+			},
+		],
+	},
+};
+
+const snap = new Snap(config);
+```
+
+
+The `initial.parameters` object takes `UrlManager` state parameters, such as filter, sort, page, and pageSize. This feature is likely to mostly be used used with filter, but any parameters including custom ones could be utilized. 
+
+There is also an optional `ignoreParameter` param you can set on the `initial.settings` object, this allows for specifying additional UrlManager state parameters to be added to the ignore list. See example where the `initial.state` filter `on_sale:yes` will be set even if there are other `filter` params present in the UrlManager state. The default values in the `ignoreParameter` are `query`, `tag`, `oq` and `fallbackQuery`. This list is used to determine wether or not to apply the initial state provided - if the UrlManager state contains any states that are not being ignored, the initial state will not be applied.
+
+```typescript
+const config = {
+	client: {
+		globals: {
+			siteId: 'abc123',
+		},
+	},
+	controllers: {
+		search: [
+			{
+				url: {
+					initial: {
+						settings: {
+							ignoreParameters: ['filter'],
+						},
+						parameters: {
+							filter: {
+								state: {
+									on_sale: ['yes'],
+								}
+							},
+						},
+					}
+				},
+				config: {
+					id: 'search',
+				},
+			},
+		],
+	},
+};
+
+const snap = new Snap(config);
+```
+
+## Advanced Configuration
+Additional advanced configuration is available for special use cases. If you do not wish to use the default `ignoreParameters` you can specify as much using `initial.settings.useDefaultIgnoreParameters`, and setting it to `false`.
+
+More configuration can be made within Within each `initial.parameter` object. It is possible to specify individual `ignoreParameters` here, which allows for more control for when to apply each individual initial state. Within the parameter configuration it is also possible to opt out of the 'global' `ignoreParameter` settings specified in `initial.settings` by using the `useGlobalIgnoreParameters` configuration. Lastly, there may be cases where if isnecessary to replace existing state values instead of merging them (the default action). This is specified in each individual parameter configuration via the `action` configuration; this supports a `merge` or `set` value.
```

Comment:
Duplicate word - "within Within"

Rephrase "there may be cases where if isnecessary to" 
to "there may be cases where it is necessary to"